connectors/sources/network_drive.py

```diff
@@ -200,9 +306,14 @@ async def get_docs(self, filtering=None):
         Yields:
             dictionary: Dictionary containing the Network Drive files and folders as documents
         """
-        directory_details = smbclient.walk(top=rf"\\{self.server_ip}/{self.drive_path}")
 
-        for path, _, _ in directory_details:
+        if filtering and filtering.has_advanced_rules():
+            advanced_rules = filtering.get_advanced_rules()
+            matched_paths, _ = self.find_matching_paths(advanced_rules)
```

Comment:
Although I don't expect any invalid rules here, because rules should be first validated before calling `get_docs`, I'd suggest adding a warning message here in case something weird happens.

Reply:
Upated in commit: [add a warning](https://github.com/elastic/connectors-python/pull/1341/commits/5dca844a1ad3bffbef0fe9fc07da2c76d15560ea)